lib/chef/knife/ec2_server_delete.rb

```diff
@@ -54,61 +61,52 @@ class Ec2ServerDelete < Knife
-          instance_id = fetch_instance_id(config[:chef_node_name])
-          @name_args << instance_id unless instance_id.nil?
-        end
-
-        @name_args.each do |instance_id|
-
-          begin
-            @server = ec2_connection.servers.get(instance_id)
-
-            msg_pair("Instance ID", @server.id)
-            msg_pair("Instance Name", @server.tags["Name"])
-            msg_pair("Flavor", @server.flavor_id)
-            msg_pair("Image", @server.image_id)
-            msg_pair("Region", ec2_connection.instance_variable_get(:@region))
-            msg_pair("Availability Zone", @server.availability_zone)
-            msg_pair("Security Groups", @server.groups.join(", "))
-            msg_pair("IAM Profile", iam_name_from_profile(@server.iam_instance_profile)) if @server.iam_instance_profile
-            msg_pair("SSH Key", @server.key_name)
-            msg_pair("Root Device Type", @server.root_device_type)
-            msg_pair("Public DNS Name", @server.dns_name)
-            msg_pair("Public IP Address", @server.public_ip_address)
-            msg_pair("Private DNS Name", @server.private_dns_name)
-            msg_pair("Private IP Address", @server.private_ip_address)
-
-            puts "\n"
-            confirm("Do you really want to delete this server")
-
-            @server.destroy
-
-            ui.warn("Deleted server #{@server.id}")
-
-            if config[:purge]
-              if config[:chef_node_name]
-                thing_to_delete = config[:chef_node_name]
-              else
-                thing_to_delete = fetch_node_name(instance_id)
-              end
-              destroy_item(Chef::Node, thing_to_delete, "node")
-              destroy_item(Chef::ApiClient, thing_to_delete, "client")
+        validate_aws_config!
+        validate_instances!
+
+        server_hashes.each do |h|
+          instance_id = h["instance_id"]
+          msg_pair("Instance ID", instance_id)
+          msg_pair("Instance Name", h["name"])
+          msg_pair("Flavor", h["instance_type"])
+          msg_pair("Image", h["image_id"])
+          msg_pair("Region", fetch_region)
+          msg_pair("Availability Zone", h["az"])
+          msg_pair("Security Groups", h["security_groups"])
+          msg_pair("IAM Profile", h["iam_instance_profile"])
+          msg_pair("SSH Key", h["key_name"])
+          msg_pair("Root Device Type", h["root_device_type"])
+          msg_pair("Public DNS Name", h["public_dns_name"])
+          msg_pair("Public IP Address", h["public_ip_address"])
+          msg_pair("Private DNS Name", h["private_dns_name"])
+          msg_pair("Private IP Address", h["private_ip_address"])
+
+          puts "\n"
+          confirm("Do you really want to delete this server")
+
+          delete_instance(instance_id) unless config[:dry_run]
+
+          ui.warn("Deleted server #{instance_id}")
+
+          if config[:purge]
+            if config[:chef_node_name]
```

Comment:
1. `thing_to_delete = config[:chef_node_name] || fetch_node_name(instance_id)`
2. can we replace this variable name as just `delete`